client/wfnews-war/src/main/angular/src/app/components/report-of-fire/review-page/rof-review-page.component.ts

```diff
@@ -1,23 +1,148 @@
+    const location = this.reportOfFire.fireLocation;
+    this.map = L.map('map', {
+      attributionControl: false,
+      zoomControl: false,
+      dragging: false,
+      doubleClickZoom: false,
+      boxZoom: false,
+      trackResize: false,
+      scrollWheelZoom: false
+    }).setView(location, 9)
+    // configure map data
+    L.tileLayer('http://{s}.google.com/vt/lyrs=s&x={x}&y={y}&z={z}', {
+      maxZoom: 20,
+      subdomains:['mt0','mt1','mt2','mt3']
+    }).addTo(this.map);
+
+    const fireLocationIcon = L.divIcon({
+      html: '<i class="fireLocationIcon material-icons">location_searching</i>',
+      iconSize: [48, 48],
+      className: 'fireLocationIcon'
+    })
+    L.marker(location, {icon:fireLocationIcon}).addTo(this.map)
+
+    L.marker( this.reportOfFire.currentLocation, {
+      icon: L.divIcon( {
+          className:  'rof-location',
+          iconSize:   [ 20, 20 ],
+          iconAnchor: [ 14, 14 ]
+      } )
+    }).addTo(this.map)
+  }
+
   initialize (data: any, index: number, reportOfFire: ReportOfFire) {
     super.initialize(data, index, reportOfFire);
+    this.reportOfFirePages = ConfigJson.pages
+    const pagesToRemove = [
+      'first-page',
+      'permissions-page',
+      'callback-page',
+      'distance-page',
+      'infrastructure-page',
+      'response-page',
+      'review-page',
+      'final-page'
+    ];
+    this.reportOfFirePages = this.reportOfFirePages.filter(page => !pagesToRemove.includes(page.id));
+  }
+
+  selectedAnswer(page:any) {
+    switch(page.id){
+      case 'contact-page' : 
+        return this.reportOfFire.consentToCall ? 'Yes' : 'No';
+      case 'location-page' :
+        return this.reportOfFire.fireLocation
+      case 'photo-page' :
+        return this.photoNumber()
+      case 'smoke-color-page' :
+        return this.reportOfFire.smokeColor;
+      case 'fire-size-page' :
+        return this.reportOfFire.fireSize;
+      case 'response-details-page' :
+        return this.reportOfFire.signsOfResponse;
+      case 'visible-flame-page' :
+        return this.reportOfFire.visibleFlame;
+      case 'fire-spread-page' :
+        return this.reportOfFire.rateOfSpread;
+      case 'what-is-burning-page' :
+        return this.reportOfFire.burning;
+      case 'infrastructure-details-page' :
+        return this.reportOfFire.assetsAtRisk;
+      case 'comments-page' :
+        return this.reportOfFire.otherInfo;
+      default :
+        return null;
+    }
+  }
+
+  selectedAnswerPart2(page: any) {
+    switch(page.id) {
+      case 'contact-page' :
+        var phoneNumber = ('' + this.reportOfFire.phoneNumber).replace(/\D/g, '');
+        var match = phoneNumber.match(/^(\d{3})(\d{3})(\d{4})$/); 
+        // reformate to phonenumber
+        if (match) {
+          return (this.reportOfFire.fullName) + '\n' + '(' + match[1] + ') ' + match[2] + '-' + match[3];
+        }
+    }
+  }
+
+  twoPartsQuestions(page:any) {
+    switch(page.id) {
```

Comment:
For single case switches, it's better to use an `if`

Reply:
thanks. Done